consensus/cometbft/service/abci.go

```diff
@@ -139,17 +143,97 @@ func (s *Service) Commit(
 	return s.commit(req)
 }
 
+// NOTE: Partially copied from https://github.com/cosmos/cosmos-sdk/blob/960d44842b9e313cbe762068a67a894ac82060ab/baseapp/abci.go#L168
+func (s *Service) Query(
+	_ context.Context,
+	req *abci.QueryRequest,
+) (*abci.QueryResponse, error) {
+	resp := &abci.QueryResponse{}
+
+	// add panic recovery for all queries
+	//
+	// Ref: https://github.com/cosmos/cosmos-sdk/pull/8039
+	defer func() {
+		if r := recover(); r != nil {
+			resp = queryResult(errorsmod.Wrapf(sdkerrors.ErrPanic, "%v", r))
+		}
+	}()
+
+	// when a client did not provide a query height, manually inject the latest
+	if req.Height == 0 {
+		req.Height = s.LastBlockHeight()
+	}
+
+	path := splitABCIQueryPath(req.Path)
+	if len(path) == 0 {
+		resp = queryResult(errorsmod.Wrap(sdkerrors.ErrUnknownRequest, "no query path provided"))
+		return resp, nil
+	}
+
+	switch path[0] {
+	case "store":
+		// "/store" prefix for store queries
+		req.Path = "/" + strings.Join(path[1:], "/")
+		cms := s.sm.GetCommitMultiStore()
+		queryable, ok := cms.(storetypes.Queryable)
+		if !ok {
+			resp = queryResult(errorsmod.Wrap(sdkerrors.ErrUnknownRequest, "multi-store does not support queries"))
+			return resp, nil
+		}
+
+		sdkReq := storetypes.RequestQuery(*req)
+		queryResp, err := queryable.Query(&sdkReq)
+		if err != nil {
+			return queryResult(err), nil
+		}
+		queryResp.Height = req.Height
+
+		resp := abci.QueryResponse(*queryResp)
+
+		return &resp, nil
+
+	default:
+		resp = queryResult(errorsmod.Wrap(sdkerrors.ErrUnknownRequest, "unknown query path"))
```

Comment:
consider refactoring this with something like
```golang
if path[0] != "store" {
   return err
}
...
```
to reduce indentation

Reply:
i had kept it like this to be a bit more inline with the original implementation, but this PR is superseded by #2737 anyways 👍